zeeguu/core/content_retriever/parse_with_readability_server.py

```diff
@@ -6,6 +6,8 @@
 
 
 def download_and_parse(url):
+    # This code will be run twice when using a newspaper source
+    # Maybe add a flag to avoid running this if the feed type == newspaper
```

Comment:
why is it running twice? how can we avoid it?

Reply:
When we crawl using newspaper we already download the articles, so it we already have the parsed information. I thought about adding a parameter to the call of the method: `feed_type`, then we can handle the fact that the newspaper text is already available.

Reply:
However, we have multiple places in the code where this is called without having the feed information naturally. So I am not sure if adding a parameter would make sense. This method is essentially asking for an URL and returning a newspaper.Article object. So it might be difficult to say that we already have that object at all times.